sasmodels/kerneldll.py

```diff
@@ -272,9 +272,11 @@ def make_dll(source, model_info, dtype=F64, system=False):
                   model_info.id, model_file, dll)
 
     if not os.path.exists(dll):
-        # Make sure the DLL path exists.
-        if not os.path.exists(SAS_DLL_PATH):
-            os.makedirs(SAS_DLL_PATH)
+        # Make sure the DLL path exists. Use abspath since python docs warn
+        # that makedirs is not robust against '..' in path.
+        os.makedirs(os.path.abspath(SAS_DLL_PATH), exist_ok=True)
+        basename = splitext(os.path.basename(dll))[0] + "_"
+        system_fd, filename = tempfile.mkstemp(suffix=".c", prefix=basename)
```

Comment:
Merge mess: these two lines are duplicated in `if not system` below.

The purpose of the @llimeht change was to create precompiled .so files for the debian distribution. I've asked him to review your change, though I think just deleting these two lines will be fine.